src/classes/ADV_Info.cls

```diff
@@ -0,0 +1,131 @@
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2017
+* @group TDTM
+* @group-content ../../ApexDocContent/TDTM.htm
+* @description Provides the Advancement Information to NPSP class.
+**/
+
+public with sharing class ADV_Info {
+    /*******************************************************************************************************
+    * @description  The boolean value to determine if Adv project is installed.
+    *******************************************************************************************************/
+    private static Boolean useAdv = false;
+
+    /*******************************************************************************************************
+    * @description  The adapter class from Adv to override the default behavior of ADV_Adapter.
+    *******************************************************************************************************/
+    private static ADV_Adapter apiNPSP = null;
+
+    /*******************************************************************************************************
+    * @description  The boolean value to determine if the setup is called before.
+    *******************************************************************************************************/
+    private static Boolean setupCalled = false;
+
+    /*******************************************************************************************************
+    * @description Setup the attributes of the class.
+    * @return void
+    *******************************************************************************************************/
+    private static void setup() {
+
+        //Check if setup() is being called. If yes, just return
+        if (setupCalled)
```

Comment:
@weizsforg  The general coding style in use throughout NPSP (at least anything built in recent years) uses opening and closing curly brackets on all code blocks even if they're just a single line:
```
if (setupCalled) {
   return;
}
```

Reply:
Thanks. I can add the curly brackets.